File: src/main/java/cn/motui/meican/order/OrderDetailPanel.java
package cn.motui.meican.order;

import cn.motui.meican.Constants;
import cn.motui.meican.model.api.vo.Progress;
import cn.motui.meican.model.ui.OrderDetail;
import cn.motui.meican.service.ServiceFactory;
import org.apache.commons.collections.CollectionUtils;
import org.apache.commons.lang3.time.DateFormatUtils;

import javax.swing.*;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

/**
 * 订单详情
 *
 * @author it.motui
 * @date 2021-01-23
 */
public class OrderDetailPanel {
  private JPanel rootPanel;
  private JList<String> scheduleList;
  private JLabel tips;
  private JLabel count;
  private JLabel title;
  private JLabel remark;

  private final LocalDateTime targetDateTime;
  private final String orderUniqueId;

  public OrderDetailPanel(LocalDateTime targetDateTime, String orderUniqueId) {
    this.targetDateTime = targetDateTime;
    this.orderUniqueId = orderUniqueId;
    this.scheduleList.setBackground(Constants.TRANSPARENT);
    this.renderUi();
  }

  private void renderUi() {
    this.scheduleList.setBackground(null);
    SwingUtilities.invokeLater(() -> {
      OrderDetail orderDetail = ServiceFactory.dataService().getOrderDetail(this.orderUniqueId);
      this.count.setText(orderDetail.getDishCount() + "份  ");
      this.title.setText(orderDetail.getDish().getTitle());
      this.remark.setText(orderDetail.getDish().getRemark());
      this.tips.setText("[" + orderDetail.getPostbox().getPostboxCode() + "] " + orderDetail.getPickUpMessage());
      List<Progress> progressList = orderDetail.getProgressList();
      if (CollectionUtils.isNotEmpty(progressList)) {
        String[] progressArray = new String[progressList.size()];
        for (int i = 0; i < progressList.size(); i++) {
          Progress progress = progressList.get(i);
          progressArray[i] = "[" + DateFormatUtils.format(new Date(progress.getTimestamp()), "HH:mm:ss") + "] "
              + progress.getActivity();
        }
        this.scheduleList.setListData(progressArray);
      }
    });
  }


  public JPanel getRoot() {
    return rootPanel;
  }
}

File: src/main/kotlin/cn/motui/meican/ui/order/OrderDetailPanel.kt
package cn.motui.meican.ui.order

import cn.motui.meican.MeiCanBundle.message
import cn.motui.meican.ui.OrderDetailForm
import cn.motui.meican.ui.UI
import cn.motui.meican.util.dataService
import cn.motui.meican.util.toLocalDateTime
import java.time.format.DateTimeFormatter
import javax.swing.JPanel
import javax.swing.SwingUtilities

/**
 * 订单详情
 */
class OrderDetailPanel constructor(
    private val orderUniqueId: String
) {

    private val form = OrderDetailForm()

    init {
        renderUi()
    }

    fun root(): JPanel {
        return form.rootPanel
    }

    private fun renderUi() {
        SwingUtilities.invokeLater {
            val orderDetail = dataService.getOrderDetail(orderUniqueId)
            form.count.text = message("order.tool.window.order.detail.count").format(orderDetail.dishCount)
            form.title.text = orderDetail.dish.title
            form.remark.text = orderDetail.dish.remark
            form.tips.text = "[" + orderDetail.postbox.postboxCode + "] " + orderDetail.pickUpMessage
            form.scheduleList.setListData(
                orderDetail.progressList.map {
                    "[" + it.timestamp.toLocalDateTime()
                        .format(DateTimeFormatter.ofPattern("HH:mm:ss")) + "] " + it.activity
                }
                    .toTypedArray()
            )
        }
    }

}